### PR TITLE
Replace the --machinefile to --machine-file

### LINKED ISSUE #25603: Command line flag `machinefile` should be `machine-file`
To match `startup-file` and `history-file`. Alternatively, the latter two could be changed. I think this has been mentioned somewhere previously but I couldn't find it. Should probably be done before 1.0.

### PR DESCRIPTION
**Pull request by srbcheema1**
It is my first PR for Julia, I want to contribute in the development of Julia Language, so just solved an easy issue.
closes #25603

### CODE REVIEW

**Comment by KristofferC** [comment]
As #25603 mentions it would be good to deprecate the original `--machinefile` so that this doesn't immediately break usages of it.

You can see an example of how this is done for another comman option at e.g. https://github.com/srbcheema1/julia/blob/1dcbfaeb65385aef2e50ec2aa5cc0a18b0c1a984/src/jloptions.c#L351-L353 where `--precompiled` was deprecated for `--sysimage-native-code`.

**Comment by StefanKarpinski** [comment]
You would probably need to keep the old `opt_machinefile` option with a deprecation warning and add a new `opt_machine_file` option.

File: doc/src/manual/getting-started.md [hunk 1]
```diff
@@ -113,7 +113,7 @@ julia [switches] -- [programfile] [args...]
 
  -p, --procs {N|auto}      Integer value N launches N additional local worker processes
                            "auto" launches as many workers as the number of local cores
- --machinefile <file>      Run processes on hosts listed in <file>
+ --machine-file <file>      Run processes on hosts listed in <file>
```
Comment (StefanKarpinski) [inline doc/src/manual/getting-started.md]: Alignment (since this gets printed out).

**Review by StefanKarpinski — APPROVED** [review]
Looks great aside from the one alignment comment.

**Comment by srbcheema1** [comment]
Thanks for the review!
I have done the changes as you suggested. Could you please take a look?

**Comment by StefanKarpinski** [comment]
I submitted a review: it looks great aside from the alignment of the help message.

**Comment by srbcheema1** [comment]
I have fixed the alignment of the help message .

**Comment by StefanKarpinski** [comment]
Ah, I see, there's another instance of it which I was still seeing in the diff. There's also a test failure which I'm looking into.

**Comment by Sacha0** [comment]
Welcome @srbcheema1! Thanks for preparing this pull request! :)

**Comment by StefanKarpinski** [comment]
The failing test is `make test-cmdlineargs` if you want to look into it, @srbcheema1. I did a bit of poking around but then the complexity of `getopt_long` and the late hour defeated me. This worked fine when you weren't adding a new option flag, so I'm guessing that some values are misaligned.

**Comment by StefanKarpinski** [comment]
Thanks for the fix, @srbcheema1!

**Comment by srbcheema1** [comment]
Thank you very much ! I learnt a lot in the process , looking forward to contribute more . :smile:

**Pull request merged**